### ISSUE TITLE
Raise a more intelligible error when trying to convert static method with name eqal to constructor

### ISSUE THREAD

**Issue by danog** [glayzzle/php-parser#86]
When trying to convert a static method ```constructor```, the parser throws an unintelligible error, `Parse Error : syntax error, unexpected 'function Object() { [native code] }', expecting T_STRING on line n`, a more specific error would be nice for JS newbies.

Example code that triggers the error:

```
<?php class a { public static function constructor() {} }
```

**Comment by danog** [comment]
The same happens when trying to call a static function with name `constructor`

`<?php anyClass::constructor();`

Or when having a normal method with name `constructor`:

`<?php class a { public function constructor() {} }`

`<?php $classinstance->constructor();` works though

**Comment by ichiriac** [comment]
Hi  @danog,

Many thanks for the bug report. Need to dig a little, but for me this is due to the fact that the function name is used/indexed somewhere as a property over an object.

I can reproduce the same behavior on :

```php
<?php class a { 
  public function __proto__() {} 
}
```

I'll work on it in few days, maybe will release it this week-end

**Comment by ichiriac** [comment]
released under 2.0.7
